pkg/auth/vault/token.go

```diff
@@ -0,0 +1,32 @@
+package vault
+
+import (
+	"github.com/IBM/argocd-vault-plugin/pkg/utils"
+	"github.com/hashicorp/vault/api"
+)
+
+// Just a plain vault token
+type TokenAuth struct {
+	AccessToken string
+}
+
+// We just want to pass-through the vault token here
+func NewTokenAuth(token string) *TokenAuth {
+	tokenAuth := &TokenAuth{
+		AccessToken: token,
+	}
+
+	return tokenAuth
+}
+
+// Authenticate authenticates with Vault and returns a token
+func (t *TokenAuth) Authenticate(vaultClient *api.Client) error {
+
```

Comment:
I think this can be updated to actually just be no-op and return nil. Because the client will use the `VAULT_TOKEN` with no intervention. So i would remove the `NewTokenAuth` method, just return nil out of `Authenticate` and you can just make `TokenAuth` an empty struct.